src/network/netdev/batadv.c

```diff
@@ -0,0 +1,197 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <inttypes.h>
+#include <linux/genetlink.h>
+
+#include "fileio.h"
+#include "batadv.h"
+#include "netlink-util.h"
+#include "network-internal.h"
+#include "networkd-manager.h"
+#include "parse-util.h"
+#include "stdio-util.h"
+#include "string-table.h"
+#include "string-util.h"
+
+static void batadv_init(NetDev *n) {
+        BatmanAdvanced *b;
+
+        b = BATADV(n);
+
+        /* Set defaults */
+        b->aggregation            = true;
+        b->gateway_bandwidth_down = 10000;
+        b->gateway_bandwidth_up   = 2000;
+        b->bridge_loop_avoidance  = true;
+        b->distributed_arp_table  = true;
+        b->fragmentation          = true;
+        b->hop_penalty            = 15;
+        b->originator_interval    = 1000;
+        b->routing_algorithm      = BATADV_ROUTING_ALGORITHM_BATMAN_V;
+}
+
+static const char* const batadv_gateway_mode_table[_BATADV_GATEWAY_MODE_MAX] = {
+        [BATADV_GATEWAY_MODE_OFF]    = "off",
+        [BATADV_GATEWAY_MODE_CLIENT] = "client",
+        [BATADV_GATEWAY_MODE_SERVER] = "server",
+};
+
+static const char* const batadv_routing_algorithm_table[_BATADV_ROUTING_ALGORITHM_MAX] = {
+        [BATADV_ROUTING_ALGORITHM_BATMAN_V]    = "BATMAN_V",
+        [BATADV_ROUTING_ALGORITHM_BATMAN_IV]   = "BATMAN_IV",
```

Comment:
hmm, we pretty exclusively use lower-casing for enum identifiers, I think we should stick to that here too. i.e. spell out as `"batman-v"` and `"batman-iv"` please

Reply:
But the routing algorithm is called "BATMAN_V" in upper case. Does it then make sense to lower case?

Reply:
This is done by intention here, as BATMAN expects exactly this strings when beeing configured via netlink.
It is used [in batadv.c](https://github.com/freifunkMUC/systemd/blob/e7f69809a6ba4c633f9d415d30e303f178c7e47c/src/network/netdev/batadv.c#L183)

Reply:
well, that's an internal detail of the kernel api. I don't think we should expose this 1:1 to users. maybe define two tables here:

batadv_routing_algorithm_table[] for the user-facing (lowercase) names, and batadv_routing_algorithm_kernel_table[] for the kernel names that use the suggested kernel spelling.

Reply:
(btw, also please align the table to be "compact", i.e. `=` should be in the same column on all rows (like you already have) but only one whitespace left and one whitepsace right of it for the "shortest" row, if you follow what i mean)

Reply:
Fixed :).